x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/search/__tests__/query_context.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { QueryContext } from '../query_context';
+import { CursorPagination } from '../..';
+import { CursorDirection, SortOrder } from '../../../../../../common/graphql/types';
+
+describe(QueryContext, () => {
+  // 10 minute range
+  const rangeStart = '2019-02-03T19:06:54.939Z';
+  const rangeEnd = '2019-02-03T19:16:54.939Z';
+
+  const pagination: CursorPagination = {
+    cursorDirection: CursorDirection.AFTER,
+    sortOrder: SortOrder.DESC,
+  };
+
+  let qc: QueryContext;
+  const makeQueryContext = () => {
+    return new QueryContext({}, rangeStart, rangeEnd, pagination, null, 10);
+  };
+  beforeEach(() => (qc = makeQueryContext()));
+
+  describe('dateRangeFilter()', () => {
+    const expectedRange = {
+      range: {
+        '@timestamp': {
+          gte: rangeStart,
+          lte: rangeEnd,
+        },
+      },
+    };
+    describe('when hasTimespan() is true', () => {
+      it('should create a date range filter including the timespan', async () => {
+        const mockHasTimespan = jest.fn();
+        mockHasTimespan.mockReturnValue(Promise.resolve(true));
+        qc.hasTimespan = mockHasTimespan;
+
+        expect(await qc.dateRangeFilter()).toEqual({
+          bool: {
```

Comment:
This type of assertion is fine with me if we all agree to use/follow it. I personally would use a snapshot for this type of response. The spirit of the snapshot is that, I have evaluated its contents and I know it's what I want in the initial write. After that it should never change unless the module's expected output changes. It results in cleaner-looking test files IMO. But this assertion is effectively the same thing as what a snapshot does, so really it's a semantic preference more than anything else. I am fine either way; if this is more helpful for evaluating what's going on in the test we should do it.

Reply:
I think my preference goes the other way. If the output of `timespanClause()` changes, this test will still work and no snapshot will need to be updated. This test isn't testing `timespanClause`, there's a dedicated test for that, so changes to the logic there shouldn't affect the pass/fail nature of the test here.

Would you disagree that this property is a benefit?

Reply:
That's a pretty good point, I do agree that in this case we don't necessarily want to test the output of functionality that's tested elsewhere.